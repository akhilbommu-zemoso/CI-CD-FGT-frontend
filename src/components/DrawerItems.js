import React from 'react'
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import Divider from '@material-ui/core/Divider'
import { Link } from 'react-router-dom'
function DrawerItems(props) {
    return (
        <Link data-testid="link" to={props.link} className={props.linksclass} onClick={() => { props.setSelected(2 ** props.index) }} >
            <ListItem
                selected={(props.selected >> props.index) & 1}
                classes={(props.selected:props.active)}>
                <Divider classes={{ root: props.dividerColor }}
                flexItem={(props.selected >> props.index) & 1}
                orientation="vertical" />
                 <ListItemIcon>{props.icon}</ListItemIcon>
                <ListItemText primary={props.itemName} />
            </ListItem>
        <Divider />
        </Link >
      
    )
}

export default DrawerItems