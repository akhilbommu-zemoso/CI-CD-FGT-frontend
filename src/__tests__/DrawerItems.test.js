import React, { Component } from 'react'
import { render, fireEvent, getAllByText, getByText, getByTestId,screen } from '@testing-library/react'
import { renderHook, act } from '@testing-library/react-hooks'
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme'
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { create } from "react-test-renderer";
import DrawerItems from '../components/DrawerItems';
import { createMemoryHistory } from "history";
Enzyme.configure({ adapter: new Adapter() });
it('should take a snapshot', () => {
   
    const { asFragment } = render(
        <Router>
        <DrawerItems  /></Router>
    )
    expect(asFragment(<Router><DrawerItems /></Router>)).toMatchSnapshot()
});

const setSelected = jest.fn();
it("visit /dashboard/home on click", () => {
    const history = createMemoryHistory();
    const { getByText,getByTestId}=  render(
        <Router history={history}>
            <DrawerItems itemName="Home" link="/dashboard/home" setSelected={setSelected}/>
        </Router>
    );
    fireEvent.click(getByTestId("link"))
    expect(window.location.href).toEqual("http://localhost/dashboard/home")
});
it("checks for drawer items", () => { 
    const { getByText, getByTestId } = render(
        <Router history={history}>
            <DrawerItems itemName="Home" link="/dashboard/home" setSelected={setSelected} />
        </Router>
    );
    expect(screen.getByText("Home")).toBeInTheDocument
})
it("checks for drawer items", () => {
    const { getByText, getByTestId } = render(
        <Router history={history}>
            <DrawerItems itemName="Wallet" link="/dashboard/wallet" setSelected={setSelected} />
        </Router>
    );
    expect(screen.getByText("Wallet")).toBeInTheDocument
})


