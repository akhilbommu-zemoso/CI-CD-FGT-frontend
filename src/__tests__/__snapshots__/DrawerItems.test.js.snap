// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <a
    data-testid="link"
    href=""
  >
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <hr
        class="MuiDivider-root MuiDivider-vertical"
      />
      <div
        class="MuiListItemIcon-root"
      />
      <div
        class="MuiListItemText-root"
      />
    </li>
    <hr
      class="MuiDivider-root"
    />
  </a>
</DocumentFragment>
`;
